presto-hive/src/main/java/com/facebook/presto/hive/metastore/SemiTransactionalHiveMetastore.java

```diff
@@ -1515,6 +1522,11 @@ private void rollbackShared()
         checkHoldsLock();
 
         for (DeclaredIntentionToWrite declaredIntentionToWrite : declaredIntentionsToWrite) {
+            // for temporary table simply delete the entire directory
```

Comment:
Not sure if we should use the extracted method as well (yeah it will need to iterate the list twice).

Reply:
I didn't want to extract that method out of the `Commiter` inner class. But now i wondering why. Let me extract it.